apps/src/p5lab/AnimationTab/AnimationList.jsx

```diff
@@ -31,21 +31,36 @@ class AnimationList extends React.Component {
     animationList: shapes.AnimationList.isRequired,
     selectedAnimation: shapes.AnimationKey,
     onNewItemClick: PropTypes.func.isRequired,
-    spriteLab: PropTypes.bool.isRequired
+    spriteLab: PropTypes.bool.isRequired,
+    hideBackgrounds: PropTypes.bool.isRequired
   };
 
   render() {
     let addAnimation = (
       <NewListItem
         key="new_animation"
         label={this.props.spriteLab ? i18n.newCostume() : i18n.newAnimation()}
-        onClick={this.props.onNewItemClick}
+        onClick={() => this.props.onNewItemClick(this.props.spriteLab)}
       />
     );
+    let animationListKeys = this.props.animationList.orderedKeys;
+    if (this.props.hideBackgrounds) {
+      animationListKeys = animationListKeys.filter(key => {
+        if (
+          this.props.animationList.propsByKey[key].categories &&
+          this.props.animationList.propsByKey[key].categories.includes(
+            'backgrounds'
+          )
+        ) {
+          return false;
+        }
+        return true;
+      });
```

Comment:
i wrote this freehand, so double-check me, but we can return the value of our conditional to simplify this filter function:

```suggestion
        return !(this.props.animationList.propsByKey[key].categories || []).includes('backgrounds')
```